app/models/manageiq/providers/autosde/inventory/parser/storage_manager.rb

```diff
@@ -48,7 +48,8 @@ def physical_storages
         :name                    => storage.name,
         :ems_ref                 => storage.uuid,
         :physical_storage_family => persister.physical_storage_families.lazy_find(storage.system_type.uuid),
-        :health_state            => storage.status
+        :health_state            => storage.status,
+        :asset_detail            => AssetDetail.new(:model => storage.system_type.name)
```

Comment:
You don't want to build an `AssetDetail` class instance here what you want is to use the `physical_storage_details` inventory collection:

```suggestion
        :asset_detail            => persister.physical_storage_details.build(:model => storage.system_type.name)
```

(You'll probably need to add this to your inventory-collections set https://github.com/ManageIQ/manageiq-providers-autosde/blob/master/app/models/manageiq/providers/autosde/inventory/persister/storage_manager.rb#L8-L20)

Reply:
```
+    add_physical_storage_details
     add_collection(storage, :physical_storage_families)
     add_collection(storage, :storage_resources)
     add_collection(storage, :storage_services)
@@ -19,4 +20,14 @@ class ManageIQ::Providers::Autosde::Inventory::Persister::StorageManager < Manag
     add_collection(storage, :ext_management_system)
     add_collection(storage, :storage_service_resource_attachments)
   end
+
+  def add_physical_storage_details
+    add_collection(storage, :physical_storage_details) do |builder|
+      builder.add_properties(
+        :model_class                  => ::AssetDetail,
+        :manager_ref                  => %i[resource],
+        :parent_inventory_collections => %i[:physical_storage]
+      )
+    end
+  end
```

Reply:
> You don't want to build an `AssetDetail` class instance here what you want is to use the `physical_storage_details` inventory collection:
> 
> ```
>         physical_storage = persister.physical_storages.build(...)
>         persister.physical_storage_details.build(:resource => physical_storage, :model => storage.system_type.name)
> ```
> 
> (You'll probably need to add this to your inventory-collections set https://github.com/ManageIQ/manageiq-providers-autosde/blob/master/app/models/manageiq/providers/autosde/inventory/persister/storage_manager.rb#L8-L20)

and I need this code snippet to be added to asset_detail inside physical storage ? (line 52)
doing so and the ui will recognize it and will present it at the right places.